File: vetor/exercicio4.c
#include <stdio.h>

int main() {
    int vetor[100];
    int maior, menor;

    printf("Digite 100 números inteiros:\n");
    for (int i = 0; i < 100; i++) {
        printf("Número %d: ", i + 1);
        scanf("%d", &vetor[i]);
    }

    maior = vetor[0];
    menor = vetor[0];

    for (int i = 1; i < 100; i++) {
        if (vetor[i] > maior) {
            maior = vetor[i];
        }
        if (vetor[i] < menor) {
            menor = vetor[i];
        }
    }

    int amplitude = maior - menor;

    printf("\nAmplitude total do conjunto: %d\n", amplitude);

    return 0;
}

File: vetor/exercicio4.py

numeros = [0] * 100

for i in range(100):
    numeros[i] = int(input(f"Digite o número {i+1}: "))

menor = numeros[0]
maior = numeros[0]

for i in range(1, 100):
    if numeros[i] < menor:
        menor = numeros[i]
    if numeros[i] > maior:
        maior = numeros[i]

amplitude = maior - menor

print(f"A amplitude total do conjunto é: {amplitude}")
